web/middlewares/permissions.go

```diff
@@ -388,3 +388,18 @@ func HasWebAppToken(c echo.Context) bool {
 	}
 	return pdoc.Type == permission.TypeWebapp
 }
+
+// GetTokenSub returns the subject of a token from the context
+func GetTokenSub(c echo.Context) (string, error) {
+	var claims permission.Claims
+	inst := GetInstance(c)
+	header := c.Request().Header.Get(echo.HeaderAuthorization)
+	token := header[len("Bearer "):]
+	err := crypto.ParseJWT(token, func(token *jwt.Token) (interface{}, error) {
```

Comment:
I'd prefer to avoid parsing the JWT and checking its validity twice. We are already doing this for checking permissions, can't we reuse that?